webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/score/ScoreService.scala

```diff
@@ -162,10 +162,14 @@ class ScoreServiceImpl(globalScoreRepository: GlobalScoreRepository, scoreReposi
 class ScoreServiceManager(readScore: ScoreService) {
 
   val handlers: Ref[List[ScoreEventHandler]] =
-    Ref.make(ComplianceScoreEventHandler :: SystemUpdateScoreHandler :: List.empty[ScoreEventHandler]).runNow
+    Ref.make(ComplianceScoreEventHandler :: List.empty[ScoreEventHandler]).runNow
 
-  def registerHandler(handler: ScoreEventHandler): UIO[Unit] = {
-    handlers.update(handler :: _)
+  def registerHandler(handler: ScoreEventHandler): IOResult[Unit] = {
+    handlers.update(handler :: _) *>
+    ZIO.ifZIO(readScore.getAll().map(_.exists(g => handler.initForScore(g._2))))(
+      handler.initEvents.flatMap(ZIO.foreach(_)(handleEvent(_))).unit,
+      ().succeed
+    )
```

Comment:
I think handle event needs to be an `UIO[Unit]. Not catching an error in it leads to unroverable errors and a non working rudder - for ex, when the init event has a problem, it rudder starts but in a bad state .